bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SelfIssueMembershipFlow.kt

```diff
@@ -3,59 +3,79 @@ package com.r3.businessnetworks.membership.flows.bno
-import com.r3.businessnetworks.membership.flows.member.RequestMembershipFlow
+import com.r3.businessnetworks.membership.flows.bno.support.BusinessNetworkOperatorFlowLogic
 import com.r3.businessnetworks.membership.flows.member.Utils.throwExceptionIfNotBNO
 import com.r3.businessnetworks.membership.states.MembershipContract
 import com.r3.businessnetworks.membership.states.MembershipState
 import com.r3.businessnetworks.membership.states.MembershipStatus
-import net.corda.core.contracts.Command
 import net.corda.core.flows.*
+import net.corda.core.identity.Party
 import net.corda.core.transactions.SignedTransaction
 import net.corda.core.transactions.TransactionBuilder
 import net.corda.core.utilities.ProgressTracker
-import java.time.Instant
 
 /**
- * Self Issue a Membership.  This can only be done by a BNO
+ * Self Issue a Membership.  This can only be done by a BNO,
+ * @param metaData BNO data needs a membership and then be activated
  */
+@Suppress("UNUSED_VARIABLE")
 @StartableByRPC
 @InitiatingFlow(version = 2)
 open class SelfIssueMembershipFlow(val metaData : Any) : FlowLogic<SignedTransaction>() {
     companion object {
-        object RequestingMembership : ProgressTracker.Step("Requesting Membership")
-        object ActivatingMembership : ProgressTracker.Step("Activating Membership")
+        object ACTIVATED_MEMBERSHIP : ProgressTracker.Step("Membership Activated")
+        object ACTIVATING_MEMBERSHIP : ProgressTracker.Step("Activating Membership")
 
         fun tracker() = ProgressTracker(
-                RequestingMembership,
-                ActivatingMembership
+                ACTIVATED_MEMBERSHIP,
+                ACTIVATING_MEMBERSHIP
         )
     }
 
     override val progressTracker = tracker()
 
     @Suspendable
-    override fun call() : SignedTransaction {
-        throwExceptionIfNotBNO(ourIdentity, serviceHub)
-
-        progressTracker.currentStep = RequestingMembership
-        val stx = subFlow(RequestMembershipFlow(ourIdentity, metaData))
-
-        val outState = stx.tx.outputs.single()
-        val membershipState = outState.data as MembershipState<*>
 
-        // If auto activate is set then don't need to call ActivateMembership
-        return if (!membershipState.isActive()) {
-            logger.info("Membership is not yet active")
-            progressTracker.currentStep = ActivatingMembership
-            val output = stx.tx.outRefsOfType(MembershipState::class.java).single()
+    override fun call(): SignedTransaction {
+        throwExceptionIfNotBNO(ourIdentity, serviceHub)
+        val configuration = serviceHub.cordaService(BNOConfigurationService::class.java)
+        val notary = configuration.notaryParty()
 
-            subFlow(ActivateMembershipFlow(output))
-        } else {
-            logger.info("Membership was automatically activated")
-            stx
+        //Start of first transaction to request a membership state
```

Comment:
you could extract both logical code blocks (i.e. the first tx and the second one) into their own methods. this would structure this method better